app/forms/questionnaire_form.py

```diff
@@ -200,15 +200,17 @@ def validate_date_range_with_period_limits_and_single_date_limits(
         period_range: timedelta,
     ) -> None:
         # Get period_limits from question
-        period_limits_item: tuple = self._get_period_limits(period_limits)
-        period_min: Mapping[str, int] = period_limits_item[0]
-        min_offset = self._get_offset_value(period_min)
+        period_limits_item = self._get_period_limits(period_limits)
+        period_min = period_limits_item[0]
 
-        # Exception to be raised if range available is smaller than minimum range allowed
-        if period_min and period_range < min_offset:
-            exception = f"The schema has invalid period_limits for {question_id}"
+        if period_min:
+            min_offset = self._get_offset_value(period_min)
 
-            raise Exception(exception)
+            # Exception to be raised if range available is smaller than minimum range allowed
+            if period_range < min_offset:
+                exception = f"The schema has invalid period_limits for {question_id}"
+
+                raise Exception(exception)
```

Comment:
Minor, but the check could be done inline to avoid the nesting. The value isn't reused either.

```py
        # Exception to be raised if range available is smaller than minimum range allowed
        if period_min and period_range < self._get_offset_value(period_min):
            exception = f"The schema has invalid period_limits for {question_id}"

            raise Exception(exception)
```

Reply:
Thought about it, but went the other way, just for readability, but equally not bothered. Next reviewer wins, thumbs up/down